Core/Core/AvoidingHelpers/Providers/AppleSingInProvider.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  AppleSingInProvider.swift
+//  Core
+//
+//  Created by Eugene Yatsenko on 10.10.2023.
+//
+
+import Foundation
+import AuthenticationServices
+import Swinject
+
+public struct AppleCredentials: Codable {
+    public var name: String
+    public var email: String
+    public var birthYear: String? = nil
```

Comment:
I guess we are not using birth year in the registration process so you can omit it.